File: pract.py
temp = 100000007


def fun(x, y, p):
    res = 1
    x = (x % p)

    if x == 0:
        return 0

    while(y > 0):

        if(y & 1):
            res = ((res*x) % p)
        y = y >> 1
        x = ((x*x) % p)

    return res

# .                              *****       Main Function       *****


t = int(input())

while t != 0:

    n = int(input())
    n=n-1
    result = fun(2, n, temp)
    print(result)

    t = t-1

File: pract.cpp
#include <iostream>
using namespace std;
int main()
{
  int t;

  cin >> t;

  while (t--)
  {
    long long int n, h, largest = 0, second_Largest = 0;

    cin >> n;
    cin >> h;

    int arr[n];

    for (int i = 0; i < n; i++) //.      array input
    {
      cin >> arr[i];

      // .       finding largest and second largest element during input
      if (arr[i] > largest)
      {
        second_Largest = largest;
        largest = arr[i];
      }
      else
      {
        if (arr[i] > second_Largest)
          second_Largest = arr[i];
      }
    }

    if (largest >= h) //.     If largest element >= health
      cout << 1 << endl;
    else if (largest + second_Largest >= h) //.         If sum of largest and second largest >= health
      cout << 2 << endl;
    else
    {
      int val = (h / (largest + second_Largest));
      int res = h - (largest + second_Largest) * val;
      if ((res - largest <= 0) || (res - second_Largest <= 0))
        cout << val * 2 + 1 << endl;
      else
        cout << val * 2 + 2;
    }
  }
  return 0;
}
